components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ChoreoApiMgtDAO.java

```diff
@@ -0,0 +1,185 @@
+     *
+     * @param connection     connection
+     * @param envUUID        UUID of the Environment
+     * @throws APIManagementException if failed to get the dataPlaneId
+     */
+    public String getDataPlaneIdForEnvironment(Connection connection, String envUUID) throws APIManagementException {
+        String dataPlaneId = null;
+        try (PreparedStatement prepStmt =
+                     connection.prepareStatement(SQLConstants.GET_DATA_PLANE_ID_FOR_ENV_SQL)) {
+            prepStmt.setString(1, envUUID);
+            try (ResultSet rs = prepStmt.executeQuery()) {
+                if (rs.next()) {
+                    dataPlaneId = rs.getString("DATA_PLANE_ID");
+                }
+            }
+        } catch (SQLException e) {
+            handleException("Failed to get the Data-Plane ID for environment UUID: " + envUUID, e);
+        }
+        return dataPlaneId;
+    }
+
+    /**
+     * Returns the Environments List for the Organization.
+     *
+     * @param organization The organization.
+     * @return List of Environments.
+     */
+    public List<Environment> getAllEnvironments(String organization) throws APIManagementException {
+
+        List<Environment> envList = new ArrayList<>();
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement prepStmt =
+                     connection.prepareStatement(SQLConstants.GET_ENVIRONMENTS_BY_ORGANIZATION_WITH_DATA_PLANE_ID_SQL))
+        {
+            prepStmt.setString(1, organization);
+            try (ResultSet rs = prepStmt.executeQuery()) {
+                while (rs.next()) {
+                    Integer id = rs.getInt("ID");
+                    String uuid = rs.getString("UUID");
+                    String name = rs.getString("NAME");
+                    String displayName = rs.getString("DISPLAY_NAME");
+                    String description = rs.getString("DESCRIPTION");
+                    String provider = rs.getString("PROVIDER");
+                    String dataPlaneId = rs.getString("DATA_PLANE_ID");
+
+                    Environment env = new Environment();
+                    env.setId(id);
+                    env.setUuid(uuid);
+                    env.setName(name);
+                    env.setDisplayName(displayName);
+                    env.setDescription(description);
+                    env.setProvider(provider);
+                    env.setDataPlaneId(dataPlaneId);
+                    env.setVhosts(ApiMgtDAO.getInstance().getVhostGatewayEnvironments(connection, id));
+                    envList.add(env);
+                }
+            }
+        } catch (SQLException e) {
+            handleException("Failed to get Environments of Organization: " + organization, e);
+        }
+        return envList;
+    }
+
+    private void handleException(String msg, Throwable t) throws APIManagementException {
+        log.error(msg, t);
+        throw new APIManagementException(msg, t);
+    }
```

Comment:
Lets just throw the exception, not log and throw.

Reply:
Fixed